.github/workflows/upload_app_store.yml

```diff
@@ -0,0 +1,63 @@
+name: Upload IPA to App Store
+on: workflow_dispatch
+jobs:
+  check_branch:
+    name: "Check branch ('release*' or 'main')."
+    runs-on: macos-11
+    steps:
+      - id: early_exit
+        if: ${{ github.ref != 'refs/heads/main' && startsWith(github.ref, 'refs/heads/release') != true }}
```

Comment:
Have you considered filtering by branch at the workflow level? That way, it will not run at all, instead of running and failing.

Reply:
Yes, but I think that a failure would provide more feedback on what is wrong, than just skipping(though I am not sure what feedback is given when workflow is skipped). 
But I can make the changes if we consider skipping to be a better option. I also wanted to keep it consistent with Android workflow.

Reply:
Because this is run manually, right? You're going to get feedback no matter what; if you do it from the UI it's going to draw a blank, if you use `gh` it will probably exit without a message. In that sense, maybe document a `gh` command line that can be used to run this from your laptop?

Reply:
Yes the branch is triggered manually. The workflow is intended to be executed **only** with a GitHub runner. It will not work on a local machine, as the workflow requires GitHub secrets.
On a local machine, Xcode can/should be used to upload a build.

Reply:
you can trigger workflows in GitHub with the `gh` command line. You can even automate it. `gh` will latch onto the workflow at GitHub and will use your secrets and everything.

Reply:
But will the workflow execution be visible in GitHub UI?
I'll have to get documented on how `gh` works. I somewhat don't like the idea of running this from local machine.

Reply:
> But will the workflow execution be visible in GitHub UI? I'll have to get documented on how `gh` works. I somewhat don't like the idea of running this from local machine.

gh is simpy a command line GitHub API client. You're going to do exactly the same you would with GitHub, only triggering it from your local machine instead of pushing buttons. Easier for automation.

Reply:
Right, sorry, I thought paths applied all over... Have you considered then requesting the branch as [an input](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs), with only those two options? Then it would only be a matter of changing the checkout step to check out that specific branch.

Reply:
hmm, I mistakenly thought that `Use workflow from branch:` meant to run the workflow on a given branch... I searched a lot how to constrain `workflow_dispatch` to specific branches only...
I will add the branch input to the workflow.

Reply:
I have ended up with something like this:
```
workflow_dispatch:
    inputs:
      branch:
        description: 'Build branch'     
        required: true
        default: 'main' 
        type: choice
        options:
          - main
          - release
```
But, it is suboptimal, since we may have multiple, short-lived release branches(for example we have 1.4.1 branch).
If we don't plan to have multiple release branches then the above solution works fine, but not otherwise. `options` field for `type: choice` is quite limited, it seems that I cannot use something like `git show-ref` to show all branches that start with `release` for example.

Current implementation with `if` statement allows building from main and any release branch.

Reply:
Perfect. Thanks for checking this out.